### PR TITLE
add support for 3.5 api

### LINKED ISSUE #918: New API changes without changelog
Some API method and objects have been added without a changelog.

What I found from `test_official`

New objects
- [x] [InputMedia](https://core.telegram.org/bots/api#inputmedia)
- [x] [InputMediaPhoto](https://core.telegram.org/bots/api#inputmediaphoto)
- [x] [InpputMediaVideo](https://core.telegram.org/bots/api#inputmediavideo)

New parameters to `Message`:
- [x] `media_group_id`

New arguments to `SendInvoice`:
- [x] `provider_data`

New method for `Bot`
- [x] [sendMediaGroup](https://core.telegram.org/bots/api#sendmediagroup)

### PR DESCRIPTION
**Pull request by Eldinnie**
As discussed do the current `InputMedia*` throw a `ValueError` when supplied with a file instead of a `file_id` or and `url`.

This is because the current workings of `InputFile` and it's mechanics to encode as multipart form data don't support multiple files. And quite frankly, I have no Idea how to do that.

Closes #918

### CODE REVIEW

**Comment by Eldinnie** [comment]
Review wanted.

File: telegram/inputmediaphoto.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        At the moment using a new file is not yet supported.
+    """
+
+    def __init__(self, media, caption=None):
+        self.type = 'photo'
+
+        if isinstance(media, PhotoSize):
+            self.media = media.file_id
+        elif hasattr(media, 'read'):
+            raise ValueError('We only support file_id or url as a valid media. Sending files is '
```
Comment (bomjacob) [inline telegram/inputmediaphoto.py]: Maybe add a TODO: about this?

File: telegram/inputmedia.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env python
```
Comment (bomjacob) [inline telegram/inputmedia.py]: Should these files maybe be in files/?

File: tests/test_inputmedia.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        assert input_media_video_dict['type'] == input_media_video.type
+        assert input_media_video_dict['media'] == input_media_video.media
+        assert input_media_video_dict['caption'] == input_media_video.caption
+        assert input_media_video_dict['width'] == input_media_video.width
+        assert input_media_video_dict['height'] == input_media_video.height
+        assert input_media_video_dict['duration'] == input_media_video.duration
+
+    def test_with_video(self, video):
+        # fixture found in test_video
+        imv = InputMediaVideo(video, caption="test 3")
```
Comment (bomjacob) [inline tests/test_inputmedia.py]: I don't think we use abbreviations in the other tests

File: tests/test_inputmedia.py [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    @flaky(3, 1)
+    @pytest.mark.timeout(10)
+    def test_send_media_group_all_args(self, bot, chat_id, media_group):
+        m1 = bot.send_message(chat_id, text="test")
+        messages = bot.send_media_group(chat_id, media_group, disable_notification=True,
+                                        reply_to_message_id=m1.message_id)
+        assert isinstance(messages, list)
+        assert len(messages) == 2
+        assert all([isinstance(mes, Message) for mes in messages])
+        assert all([mes.media_group_id == messages[0].media_group_id for mes in messages])
```
Comment (bomjacob) [inline tests/test_inputmedia.py]: Maybe a TODO: about needing to test with new files.

File: tests/test_official.py [hunk 5]
```diff
@@ -107,6 +107,8 @@ def check_object(h4):
             field = 'from_user'
         elif name.startswith('InlineQueryResult') and field == 'type':
             continue
+        elif name.startswith('InputMedia') and field == 'type':
```
Comment (bomjacob) [inline tests/test_official.py]: Could you combine with above? 
```elif (name.startswith('InlineQueryResult') or name.startswith('InputMedia')) and field == 'type':```
seems cleaner to me.

File: tests/test_photo.py [hunk 6]
```diff
@@ -174,9 +174,9 @@ def test_send_bytesio_jpg_file(self, bot, chat_id):
         assert isinstance(photo.file_id, str)
         assert photo.file_id != ''
         assert isinstance(photo, PhotoSize)
-        assert photo.width == 1920
-        assert photo.height == 1080
-        assert photo.file_size == 30907
+        assert photo.width == 1280
```
Comment (bomjacob) [inline tests/test_photo.py]: What changed here?
And how... like it's a smaller image but it takes up more space? :/

File: telegram/inputmedia.py (hunk 2 above)
Comment (Eldinnie) [reply telegram/inputmedia.py]: Hmm, maybe.

File: telegram/inputmediaphoto.py (hunk 1 above)
Comment (Eldinnie) [reply telegram/inputmediaphoto.py]: yes, will do

File: tests/test_inputmedia.py (hunk 3 above)
Comment (Eldinnie) [reply tests/test_inputmedia.py]: We don;t and I'm lazy, will fix

File: tests/test_inputmedia.py (hunk 4 above)
Comment (Eldinnie) [reply tests/test_inputmedia.py]: yup

File: tests/test_official.py (hunk 5 above)
Comment (Eldinnie) [reply tests/test_official.py]: yes, I pondered on it, but I will

File: tests/test_photo.py (hunk 6 above)
Comment (Eldinnie) [reply tests/test_photo.py]: I have no Idea, but it's what came back...

**Comment by Eldinnie** [comment]
I addressed everything @bomjacob pointed out. As discussed I think we should merge and release without the option to send new files yet.

Appveyor will fail due to python 3.3 drop in support from pytest. As will we (see #930) Good to go?

File: tests/test_official.py [hunk 7]
```diff
@@ -105,9 +105,8 @@ def check_object(h4):
         field = parameter[0]
         if field == 'from':
             field = 'from_user'
-        elif name.startswith('InlineQueryResult') and field == 'type':
-            continue
-        elif name.startswith('InputMedia') and field == 'type':
+        elif (name.startswith('InlineQueryResult') or name.startswith('InputMedia')) \
```
Comment (bomjacob) [inline tests/test_official.py]: Could you wrap it all in paranthesis instead of using the backslash here maybe?, doesn't look very clean IMO

Comment (Eldinnie) [reply tests/test_official.py]: 👍

**Pull request merged**